File: readability.py
from cs50 import get_string

text = get_string("Text:")

letter=0
word=1
sent=0


#One fish. Two fish. Red fish. Blue fish.
for i in range(len(text)):
    if (text[i]>= 'a' and text[i]<='z') or (text[i]>='A' and text[i]<='Z'):
     letter+=1
  
    elif text[i]== ' ':
     word+=1
  
    elif text[i]=='.' or text[i]=='?' or text[i]=='!':
     sent+=1



grade = round(0.0588*(100*letter/word)-0.296*(100* sent/word)-15.8)
if grade<1:
    print("Before Grade 1")
elif grade < 16:
    print(f"Grade {grade}")
else :
    print("Grade 16+");
    
 

File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main (void){
 string text = get_string("Text: ");
 int lettercount =0;
 int wordcount = 1;
 int sentencecount = 0;

 //count word
 for (int i=0; i < strlen(text); i++)
 {
 if ((text[i]>= 'a' && text[i]<='z') || (text[i]>='A' && text[i]<='Z')){
 lettercount++;
  }
  else if (text[i]== ' '){
  wordcount++;
  }
  else if (text[i]=='.' || text[i]=='?' || text[i]=='!'){
  sentencecount++;
  }
  }
 //index = 0.0588 * L - 0.296 * S - 15.8
 float grade = 0.0588*( 100*(float)lettercount/ (float) wordcount)-0.296*(100*(float) sentencecount/(float) wordcount)-15.8;
 if (grade< 16 && grade >= 0){
  printf("Grade %i\n", (int) round (grade));
 }
  else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }
 }
